nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/AbstractCommentManager.java

```diff
@@ -160,4 +162,38 @@ protected void setCommentPermissions(CoreSession session, DocumentModel document
         return ancestorIds;
     }
 
+    /**
+     * @since 11.1
+     */
+    protected DocumentModel getThreadForComment(CoreSession coreSession, DocumentModel comment)
+            throws CommentSecurityException {
+
+        NuxeoPrincipal principal = coreSession.getPrincipal();
+        return CoreInstance.doPrivileged(coreSession, session -> {
+            DocumentModel thread = comment;
+            DocumentModel parent = session.getDocument(new IdRef((String) thread.getPropertyValue(COMMENT_PARENT_ID)));
+            if (parent.hasSchema(COMMENT_SCHEMA)) {
+                thread = getThreadForComment(parent);
+            }
+            DocumentRef ancestorRef = session.getDocument(new IdRef((String) thread.getPropertyValue(COMMENT_PARENT_ID)))
+                                       .getRef();
+            if (!session.hasPermission(principal, ancestorRef, SecurityConstants.READ)) {
+                throw new CommentSecurityException("The user " + principal.getName()
+                        + " does not have access to the comments of document " + ancestorRef.reference());
+            }
+            return thread;
+        });
+    }
+
+    @Override
+    public DocumentRef getAncestorRef(CoreSession coreSession, DocumentModel documentModel) {
```

Comment:
This method doesn't need to be there as you're overriding it in `CommentManagerImpl`, probably the same for `getThreadForComment`.

Furthermore, public API should be with id, otherwise we will have same issues than previous API.

Getting the last elements of `comment:ancestorId` didn't work?

Reply:
Getting the last elements of comment:ancestorId didn't work? ->> for the `CommentManagerImpl`  didn't work, it's why we provide the two implementations.

Reply:
We're talking about implementation for PropertyCommentManager but on code present in the abstract. You should move it back to the correct class.
After that, does the `comment:ancestorId` way (proposed by @NourNuxeo or @troger) works for property comment manager?